lib/graphql/client.ts

```diff
@@ -37,4 +60,109 @@ export class GraphQLClient {
       throw new GraphQLError(serializeGraphQLError(error));
     }
   };
+
+  /**
+   * Return subscription client connection params
+   */
+  private getConnectionParams = (): ConnectionParams => ({
+    Authorization: `Bearer ${this.auth.tokenManager.token?.accessToken}`
+  });
+
+  /**
+   * Create a subscription client
+   */
+  private createSubscriptionClient = (): SubscriptionClient => {
+    if (!this.auth.tokenManager.token) {
+      throw new UserUnauthorizedError();
+    }
+
+    const WebSocket = typeof window === "undefined" ? ws : window.WebSocket;
+
+    return new SubscriptionClient(
+      this.subscriptionEndpoint,
+      {
+        lazy: true,
+        connectionParams: this.getConnectionParams
+      },
+      WebSocket
+    );
+  };
+
+  /**
+   * Handle disconnection:
+   *
+   * 1. Refresh access token
+   * 2. Destroy existing subscription client
+   * 3. Resubscribe all previously active subscriptions
+   */
+  private handleDisconnection = async (): Promise<void> => {
+    await this.auth.tokenManager.refresh();
+    this.subscriptionClient = null;
+    Object.values(this.subscriptions).forEach(subscription => {
+      this.subscribe(
+        subscription.query,
+        subscription.type,
+        subscription.handler,
+        subscription.id
+      );
+    });
+  };
+
+  /**
+   * Subscribe to a topic and call the handler when new data or an error is received
+   */
+  public subscribe = (
+    query: string,
+    type: SubscriptionType,
+    handler: Function,
+    subscriptionId?: number
+  ): Unsubscribe => {
+    if (!this.subscriptionClient) {
+      this.subscriptionClient = this.createSubscriptionClient();
+
+      this.subscriptionClient.onDisconnected(this.handleDisconnection);
```

Comment:
This can lead to a `UnhandledPromiseRejectionWarning` as `handleDisconnection` is async and its rejections (e.g. error from `await this.auth.tokenManager.refresh();`) are not catched anywhere. To reproduce one can just stop the backend while a subscription is active.

Reply:
Great point, see my below comment regarding that, changes are in this commit: 67de95da193ac9c4b86712c10132d47da7da22a6